gravitee-apim-gateway/gravitee-apim-gateway-handlers/gravitee-apim-gateway-handlers-api/src/main/java/io/gravitee/gateway/handlers/api/ApiContextHandlerFactory.java

```diff
@@ -78,23 +76,28 @@
 
     private final Logger logger = LoggerFactory.getLogger(ApiContextHandlerFactory.class);
 
-    @Autowired
     private ApplicationContext applicationContext;
-
-    @Value("${reporters.logging.max_size:-1}")
-    private int maxSizeLogMessage;
-
-    @Value("${reporters.logging.excluded_response_types:#{null}}")
-    private String excludedResponseTypes;
-
-    @Value("${handlers.request.headers.x-forwarded-prefix:false}")
-    private boolean overrideXForwardedPrefix;
-
-    @Value("${classloader.legacy.enabled:true}")
-    private boolean classLoaderLegacyMode;
-
-    @Autowired
-    private Node node;
+    private final int maxSizeLogMessage;
+    private final String excludedResponseTypes;
```

Comment:
I'm definitely not a big fan of that sort of approach which impies to change the constructor each time you will have a new configuration property to manage.

Reply:
yes, I was wondering if it worth creating a `ReactorHandlerFactoryProperties` to wrap all these properties into a dedicated object
wdyt ?

Reply:
I now use the Gateway Node Configuration object directly.